preview_generator/preview/builder/pdf__pypdf2.py

```diff
@@ -108,15 +109,20 @@ def get_page_number(
         cache_path: str,
         mimetype: typing.Optional[str] = None,
     ) -> int:
-        with open(cache_path + preview_name + '_page_nb', 'w') as count:
-            count.seek(0, 0)
-
-            with open(file_path, 'rb') as doc:
-                inputpdf = utils.get_decrypted_pdf(doc)
-                count.write(str(inputpdf.numPages))
-        with open(cache_path + preview_name + '_page_nb', 'r') as count:
-            count.seek(0, 0)
-            return int(count.read())
+        if not os.path.exists(cache_path + preview_name + '_page_nb'):
+            with open(cache_path + preview_name + '_page_nb', 'w') as count:
+                count.seek(0, 0)
+                with open(file_path, 'rb') as doc:
+                    inputpdf = utils.get_decrypted_pdf(doc)
+                    num_page = inputpdf.numPages
+                    count.write(str(num_page))
+                    return int(num_page)
+        else:
```

Comment:
I would invert two cases (if os.path.exists() as first case). @inkhey @buxx what do you think?